### PR TITLE
magento/magento2#22317: PR#22321 fix.

### PR DESCRIPTION
**Pull request by p-bystritsky**
### Description (*)

magento/magento2#22321 was merged before it was finished.

### Fixed Issues (if relevant)

1. magento/magento2#22317: CodeSniffer should not mark correctly aligned DocBlock elements as code style violation.

### Manual testing scenarios (*)

1. Edit any Magento 2 class - align method params like shown here.
2. Create two aligned param comments for one method (one of them should be a substring of relevant param name).
3. Run static tests.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: dev/tests/static/framework/Magento/Sniffs/Annotation/MethodArgumentsSniff.php [hunk 1]
```diff
@@ -124,8 +123,7 @@ private function getMethodArguments(File $phpcsFile, int $openParenthesisPtr, in
     private function getMethodParameters(array $paramDefinitions): array
     {
         $paramName = [];
-        $paramCount = count($paramDefinitions);
-        for ($i = 0; $i < $paramCount; $i++) {
+        foreach (array_keys($paramDefinitions) as $i) {
```
Comment (orlangur) [inline dev/tests/static/framework/Magento/Sniffs/Annotation/MethodArgumentsSniff.php]: `array_keys` is not a good idea. Why cannot use previous implementation? If there are gaps in keys, use `foreach` and ignore values.

Comment (orlangur) [reply dev/tests/static/framework/Magento/Sniffs/Annotation/MethodArgumentsSniff.php]: Same for all other changed loops.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
@p-bystritsky next time please mark PR as Draft/WiP until it is intended to be merged.

**Comment by p-bystritsky** [comment]
@orlangur updated.

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Pull request merged**